cpp/src/arrow/util/future.h

```diff
@@ -840,6 +840,18 @@ inline Future<>::Future(Status s) : Future(internal::Empty::ToResult(std::move(s
 ARROW_EXPORT
 Future<> AllComplete(const std::vector<Future<>>& futures);
 
+
+/// \brief Create a Future which completes when all of `futures` complete.
+///
+/// The future will be marked complete if all `futures` complete
+/// successfully. Otherwise, it will be marked failed with the status of
+/// the first failing future.
```

Comment:
nit: to be clear, this could be "…status of a failing future" (since "first failing future" above implies the first one to fail, not the first one in the list).

Reply:
same here